### PR TITLE
Port over and update 'Streamline aim and fire experience (#59977)'

### PR DESCRIPTION
**Pull request by KheirFerrum**
#### Summary

SUMMARY: Interface "Port over 'Streamline aim and fire experience (#59977)' "

#### Purpose of change

Ports over [#59977](https://github.com/CleverRaven/Cataclysm-DDA/pull/59977).

Fixes issues with RELOAD_AND_SHOOT weapons described in [#54997](https://github.com/CleverRaven/Cataclysm-DDA/issues/54997). No need to fix issue described in [#50571](https://github.com/CleverRaven/Cataclysm-DDA/issues/50571) as it is already fixed in BN

#### Describe the solution

Allows re-entering of aim UI immediately after firing so long as there are still targets. Option is default on but can be turned off from the settings menu under Interface.

Add "reload_time" to `aim_activity_actor` as a member variable, which will save moves needed to load weapon. This will allow it to be refunded if the player starts the aim activity then immediately exits. This cost will then be referenced in the aiming UI. Displayed separately from the firing cost. If the "first turn" has passed, the firing cost is triggered and the unload cost is shown instead.

#### Testing

- [x] Spawn rifle, test that when firing rifle at targets, we return to the UI, check that it doesn't do anything stupid when it runs out of ammo, whether we have appropriate ammo/magazines or not.
- [x] Repeat test with revolver and similar weapons with integral magazines.
- [x] Repeat test with bow and crossbow, using multiple different types of ammo.
- [x] Ensure `first_turn` is reset and removed appropriately during each cycle.

#### Additional context

~Not sure if we want to add unload cost to the menu as well. It might be useful for players to know.~
- Added.

### CODE REVIEW

File: src/ranged.cpp [hunk 1]
```diff
@@ -3098,6 +3127,11 @@ bool target_ui::action_aim()
         do_aim( *you, *relevant, min_recoil );
     }
 
+    if( activity->first_turn ) {
+        you->moves -= activity->reload_time;
+        activity->first_turn = false;
+    }
```
Comment (Coolthulhu) [inline src/ranged.cpp:3133]: I don't like this bit at all. It's "realistic", but it won't be communicated well and is unintuitive.
Instead, it's better to have the cost be added to cost of firing and only that. Having RAS weapons go through normal reload function and apply all the associated move costs immediately is a mistake.

Comment (KheirFerrum) [reply src/ranged.cpp:3133]: The action cost of loading can be two turns easily though, I'm concerned about this forcing players to commit to a rather long action in a tense situation. If we're going to have this cost at all, at least if you decide to aim you're likely in a "relatively" safe position.

**Comment by olanti-p** [comment]
Ammo type doesn't show up anymore when aiming with a RAS weapon
image

**Review by olanti-p — COMMENTED** [review]
Found a neat interaction: you can shoot your bow, then walk over to the arrow, then shoot the bow again without explicitly picking up arrow first.

**Comment by KheirFerrum** [comment]
> Found a neat interaction: you can shoot your bow, then walk over to the arrow, then shoot the bow again without explicitly picking up arrow first.

You could always do this I recall, it just has to be within 1 tile.

**Review by olanti-p — APPROVED** [review]
Ok, this PR is a definite improvement overall.
Drastically cuts down on the keypresses, and spaghetti removal is always good.

**Pull request merged**